### ISSUE TITLE
openssl issue after using mysql2

### ISSUE THREAD

**Issue by gingerlime** [brianmario/mysql2#944]
We reported this on https://github.com/pat/thinking-sphinx/issues/1087 but it seems to be related to the mysql2 gem, under [certain conditions...](https://stackoverflow.com/questions/49035866/very-strange-behaviour-with-ruby-openssl-unicorn-systemd-gcloud)

In short, if we do something like

```ruby
# this works
OpenSSL::Digest::SHA256.new
client = Mysql2::Client.new(
  :flags           => Mysql2::Client::MULTI_STATEMENTS,
  :connect_timeout => 5,
  :host            => "127.0.0.1",
  :port            => 9306
)
results = client.query("SELECT * FROM article_core").to_a
# this fails (line 33)
OpenSSL::Digest::SHA256.new
```

We get
```
RuntimeError (Unsupported digest algorithm (SHA256).)
        /usr/lib/ruby/2.3.0/openssl/digest.rb:40:in `initialize'
        /usr/lib/ruby/2.3.0/openssl/digest.rb:40:in `block (3 levels) in <class:Digest>'
        /var/local/app/app/controllers/home_controller.rb:33:in `new'
```

And any request to OpenSSL afterwards fails with similar problems...

**Comment by gingerlime** [comment]
Thanks @pat. Great ideas/investigation. I'll try requiring openssl/digest.

We're still on ruby 2.3 unfortunately ... Tested with both 2.3.4 and 2.3.6. We can try upgrading though and see if it makes a difference (last time we bumped into issues with some gems if I recall, and we simply postponed this).

**Comment by gingerlime** [comment]
further to [Jasim's comment](https://github.com/pat/thinking-sphinx/issues/1087#issuecomment-369998687), it looks like the problem just vanished. We were testing and reproducing it *consistently*, across *several* Compute Engine instances on Google Cloud. Under certain conditions (unicorn / puma started by systemd, etc), it was completely reproducible both with our own rails app, and with a plain vanilla rails app we've set for testing purposes. It was reproducible across several ruby versions as well (we tested 2.3.4, 2.3.6 and 2.5.0).

Suddenly, *all instances* started working without exhibiting these problems. Like it never existed. We didn't even reboot those instances, and we saw no evidence of any unattended upgrades taking place... We also had one snapshot of a system that had this problem, and that we can reliably reproduce on. Creating an instance from this snapshot stopped exhibiting it as well from that specific point in time a few hours ago.

We're totally confused as to what might have cause it, and what might have made it disappear... However, without being able to reproduce it now, I guess there's no point leaving this issue open, so will close it. Chalk it up to *Deus ex machina* I suppose. (perhaps the google support gods, but they haven't reported anything back to us yet)
